pilot/pkg/model/service.go

```diff
@@ -492,6 +490,18 @@ type IstioEndpoint struct {
 
 	// If in k8s, the node where the pod resides
 	NodeName string
+
+	// precomputedEnvoyEndpoint is a cached LbEndpoint, converted from the data, to
+	// avoid recomputation
+	precomputedEnvoyEndpoint atomic.Pointer[*endpoint.LbEndpoint]
+}
+
+func (ep *IstioEndpoint) EnvoyEndpoint() *endpoint.LbEndpoint {
+	return ptr.OrEmpty(ep.precomputedEnvoyEndpoint.Load())
+}
+
+func (ep *IstioEndpoint) ComputeEnvoyEndpoint(now *endpoint.LbEndpoint) {
+	ep.precomputedEnvoyEndpoint.Store(&now)
```

Comment:
I am not sure why  not store the address  of  LbEndpoint directly

Reply:
So we can atomically update it without a big mutex around the entire shard

Reply:
I mean we can ep.precomputedEnvoyEndpoint.Store(now) instead, you are storing double pointer